### DISCUSSION TITLE
Time-Series DB

### DISCUSSION THREAD

**Discussion by beriberikix** [tursodatabase/libsql#912] [Ideas]
There are a lot of applications where a simple Time-Series DB would be ideal if it was embedded/local. I work in IoT and it's extremely common to try to run something like Timescale or Influx on the edge (Influx even launched a [product](https://www.influxdata.com/edge-influxdb-cloud/) last year.)

So what's needed to bring TS to libSQL? Setting up the [table efficiently](https://stackoverflow.com/questions/65422890/how-to-use-time-series-with-sqlite-with-fast-time-range-queries), [time-related queries](https://docs.influxdata.com/influxdb/v2/query-data/common-queries/iot-common-queries/) and efficient [compression](https://www.timescale.com/blog/time-series-compression-algorithms-explained/) (ex.gorilla) come to mind but are there other challenges to supporting Time-Series data?

**Comment by franchb** [comment]
[Stanchion – Column-oriented tables in SQLite](https://github.com/dgllghr/stanchion)

**Comment by awksedgreep** [comment]
Have a look at timeless-libsql if you are still interested.

**Comment by awksedgreep** [comment]
We've moved well past gorilla these days.  I use openzl style compression(modified version) for logs / traces and modified PCO in rust for metrics.  Hope you like it.  We generally get 10:1, sometimes 12:1 or higher.  Traces are sometimes a bit lower.
